server/models/Collection.test.js

```diff
@@ -416,3 +418,63 @@ describe("#membershipUserIds", () => {
     expect(membershipUserIds.length).toBe(6);
   });
 });
+
+describe("#findByPk", () => {
+  test("should return collection with collection Id", async () => {
+    const { collection } = await seed();
+    const response = await Collection.findByPk(collection.id);
+
+    expect(response.id).toBe(collection.id);
+  });
+
+  test("should return collection when urlId is present", async () => {
+    const { collection } = await seed();
+    const id = `${slugify(collection.name)}-${collection.urlId}`;
+
+    const response = await Collection.findByPk(id);
+
+    expect(response.id).toBe(collection.id);
+  });
+
+  test("should return undefined when incorrect uuid type", async () => {
+    const { collection } = await seed();
+    const response = await Collection.findByPk(collection.id + "-incorrect");
+
+    expect(response).toBe(undefined);
+  });
+
+  test("should return undefined when incorrect urlId length", async () => {
+    const { collection } = await seed();
+    const id = `${slugify(collection.name)}-${collection.urlId}incorrect`;
+
+    const response = await Collection.findByPk(id);
+
+    expect(response).toBe(undefined);
+  });
+
+  test("should return null when uuid isn't present", async () => {
+    const { collection } = await seed();
+    const response = await Collection.findByPk("a" + collection.id.slice(1));
+
+    expect(response).toBe(null);
+  });
+
+  test("should return null when urlId isn't present", async () => {
+    const id = `${slugify("not present collection")}-${randomstring.generate(
+      15
+    )}`;
+
+    const response = await Collection.findByPk(id);
+
+    expect(response).toBe(null);
+  });
+
+  test("should return null when urlId is present but corresponding collection name is incorrect", async () => {
+    const { collection } = await seed();
+    const id = `${slugify("not present collection")}-${collection.urlId}`;
+
+    const response = await Collection.findByPk(id);
+
+    expect(response).toBe(null);
+  });
```

Comment:
Hmm, this is unexpected behavior – as long as the `urlId` is correct the slug portion can be anything. This enables the title to be changed and existing links that might exist to continue working

Reply:
Noted. 👍🏻